src/modules/roc_netio/target_uv/roc_netio/transceiver.cpp

```diff
@@ -152,7 +124,15 @@ bool Transceiver::add_udp_receiver(packet::Address& bind_address,
 
     run_task_(task);
 
-    return task.result;
+    if (task.flags & Task::PortError) {
+        if (task.flags & Task::PortAdded) {
+            wait_port_removed_(bind_address);
```

Comment:
This approach with flags looks a bit dirty. What do you think about replacing the `flags` field with `bool success` and `IElement* element` fields and using them like this?

```
if (!tesk.success) {
    if (task.element) {
        wait_element_removed_(task.element);
    }
}

return task.success;
```